justfile

```diff
@@ -7,5 +7,15 @@ default:
 build:
     docker build . -t research-template
 
-smoke-test:
-    docker run --rm research-template ls /opt/venv/bin/python3.10
\ No newline at end of file
+check-image-exists:
+    # Extra brackets are for Just escaping.
+    docker image inspect --format='{{{{.Id}}}}' research-template
```

Comment:
If you do `docker run` and the image doesn't exist, Docker will tell you.

However, Docker will then immediately try and pull an arbitrary `research-template` image from the internet, which isn't at present registered, but someone could do so. There is no way to disable this.

I dislike this behaviour a lot. So this is a preventative measure to ensure that we only run a local image.

The alternative would be to refer to a local image with a name that we do control like  `ghcr.io/opensafely-core/research-template` and give it a `dev` version.

Reply:
As elsewhere, I'm not too tied to this specifically, but I'm inclined to merge this now as it's been open a while, and get the benefit of having the tests running.

If we decide this could work better, this is OK, and would be a reasonable follow-up PR.

Reply:
Thanks for clarifying. That is, indeed, very dislikable behaviour.